lib/Optimizer/GraphOptimizer/Lower.cpp

```diff
@@ -651,6 +651,58 @@ static void lowerGroupConvolutionNode(Function *F, CompilationContext &cctx,
   replaceAllUsesOfWith(cctx.loweredInfoMap, BNG.getResult(), result);
 }
 
+static void lowerBucketizeNode(Function *F, CompilationContext &cctx,
+                               const BucketizeNode &B) {
+  // Bucketize is:
+  // Boundaries[i-1] < x <= Boundaries[i]
+  // If 'x' is beyond the bounds of Boundaries,
+  // 0 or len(Boundaries) is returned as appropriate.
+  // The node is lowered as follows:
+  // 1. For each value node in the input, broadcast it #buckets
+  // 2. Compare the broadcasted value to the buckets
+  // 3. Count the number of buckets smaller than the current value:
+  // 3.1 If they are all bigger = output is zero
+  // 3.2 Else if they are all smaller = output is len(Boundaries)
+  // 3.2 Else output is Boundaries[i-1] < x <= Boundaries[i]
+  // 4. Gather the all 'x's and create a new tensor to replace the node with
+  auto numOfBuckets = B.getBoundaries().size();
+  std::string baseStr = B.getName().str();
+  auto *boundariesConst = F->getParent()->createConstant(
+      ElemKind::FloatTy, {numOfBuckets}, baseStr + ".const");
+  boundariesConst->getPayloadMutable().getHandle<float>() = B.getBoundaries();
+  auto *zeroSplat =
+      F->createSplat("zeroSplat", boundariesConst->getType(), 0.0);
+  auto *oneSplat = F->createSplat("oneSplat", boundariesConst->getType(), 1.0);
+  auto numOfInputs = B.getInput().dims()[0];
+  auto numOfDims = B.getInput().dims().size();
+  auto *reshapredInput = F->createReshape(
+      baseStr + ".reshape.input", B.getInput(), {numOfInputs * numOfDims});
```

Comment:
I'm not sure I understand this. I think we probably want to reshape to `{B.getInput().size()}`, i.e. 1D with the number of elements, no? The op is agnostic of shape.